### PR TITLE
html-css-popup

### PR DESCRIPTION
**Pull request by yevheniiii**
### HTML-CSS-POPUP

[Demo](https://yevheniiii.github.io/) |
[Code base](https://github.com/yevheniiii/frontend-2019-p2p/tree/html-css-popup/submissions/yevheniiii/html_css_popup)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/yevheniiii/html_css_popup/index.html [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+<head>
+    <title>Kottans Pop-up</title>
+    <meta charset="UTF-8">
+    <link rel="stylesheet" href="css/main.css" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+</head>
+
+<body>
+    <nav>
+        <section class="menu">
```
Comment (A-Ostrovnyy) [inline submissions/yevheniiii/html_css_popup/index.html]: Good practice to use list structure for menu.
Use `ul > li` istead of `section > section`.
Read more about HTML tag `<section>` pay attention to its semantics.

File: submissions/yevheniiii/html_css_popup/index.html [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                        </a>
+                    </li>
+
+                    <li class="inside-item space"></li>
+                </ul>
+
+                <input type="checkbox" id="buttonMore" class="MoreApps" />
+                <label for="buttonMore" class="MorePopUp">More</label>
+
+                <hr class="line" />
```
Comment (A-Ostrovnyy) [inline submissions/yevheniiii/html_css_popup/index.html]: For line better use elements border. 
Avoid to clogging up the markup by unnecessary tags.

File: submissions/yevheniiii/html_css_popup/index.html [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+            <section class="nav-item">
+                <a><img src="images/bell.png" alt="user-notifications" /></a>
+            </section>
+
+            <section class="nav-item">
+                <a><img class="user-avatar" src="images/img_avatar.png" alt="user-avatar" /></a>
+            </section>
+
+            <section class="popup-triangle">
```
Comment (A-Ostrovnyy) [inline submissions/yevheniiii/html_css_popup/index.html]: Use CSS for creating triangles.

File: submissions/yevheniiii/html_css_popup/index.html [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                <section class="empty"></section>
+            </section>
+        </section>
+    </nav>
+</body>
+
+</html>
```
Comment (A-Ostrovnyy) [inline submissions/yevheniiii/html_css_popup/index.html]: Add an empty line at the end of each file.

File: submissions/yevheniiii/html_css_popup/css/main.css [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    width: 0;
+    height: 0;
+    border-left: 10px solid transparent;
+    border-right: 10px solid transparent;
+    border-bottom: 10px solid #fff;
+}
+
+.pop-up,
+.MoreApps {
+    position: absolute;
```
Comment (A-Ostrovnyy) [inline submissions/yevheniiii/html_css_popup/css/main.css]: Watch this [video ](https://www.youtube.com/watch?v=E6kLaaQFctU&t=413s)to find out how to hide checkbox in right way

File: submissions/yevheniiii/html_css_popup/css/main.css [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    top: 2px;
+    left: -10px;
+    width: 0;
+    height: 0;
+    border-left: 10px solid transparent;
+    border-right: 10px solid transparent;
+    border-bottom: 10px solid #fff;
+}
+
+.pop-up,
```
Comment (A-Ostrovnyy) [inline submissions/yevheniiii/html_css_popup/css/main.css]: Add styles for `:focus` state for all interactive elements.

File: submissions/yevheniiii/html_css_popup/index.html (hunk 1 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/index.html]: Thank you! I changed this.

File: submissions/yevheniiii/html_css_popup/index.html (hunk 2 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/index.html]: I removed the tag from HTML and added it as element border. Thanks!

File: submissions/yevheniiii/html_css_popup/index.html (hunk 3 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/index.html]: Thank you for the comment. Did it.

File: submissions/yevheniiii/html_css_popup/index.html (hunk 4 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/index.html]: I did this. 

Can you explain why we should add an empty line?
@A-Ostrovnyy

File: submissions/yevheniiii/html_css_popup/css/main.css (hunk 5 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/css/main.css]: Huh, never heard about this approach. Thank you very much! @A-Ostrovnyy

File: submissions/yevheniiii/html_css_popup/css/main.css (hunk 6 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/css/main.css]: Done.

File: submissions/yevheniiii/html_css_popup/index.html (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/yevheniiii/html_css_popup/index.html]: [Read more](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline?noredirect=1&lq=1)

**Comment by A-Ostrovnyy** [comment]
Good work!
All these images must have styles on `:focus` At this moment only the first two have it.
image

Also, button `More` inside popup without styles on `:focus` and `:hover` state. Add `cursor: pointer;` on it to show to the user an interactive element.

**Comment by yevheniiii** [comment]
> Good work!
> All these images must have styles on `:focus` At this moment only the first two have it.
> image
> 
> Also, button `More` inside popup without styles on `:focus` and `:hover` state. Add `cursor: pointer;` on it to show to the user an interactive element.

File: submissions/yevheniiii/html_css_popup/index.html (hunk 4 above)
Comment (yevheniiii) [reply submissions/yevheniiii/html_css_popup/index.html]: Shame on me that I didn't find it on Google.
Thanks!

**Comment by A-Ostrovnyy** [comment]
Check your markup using [W3CValidator](https://validator.w3.org/) and fix all errors.
Also, check the order of focused elements [Video](https://share.vidyard.com/watch/cphCekJshfvq6Ys8sNngVs?)
Still can't see styles on `focus` state on the button More, and app icon.

**Comment by yevheniiii** [comment]
> Check your markup using [W3CValidator](https://validator.w3.org/) and fix all errors.
> Also, check the order of focused elements [Video](https://share.vidyard.com/watch/cphCekJshfvq6Ys8sNngVs?)
> Still can't see styles on `focus` state on the button More, and app icon.

@A-Ostrovnyy thank you for your review!

1) markup is w/o errors - https://monosnap.com/file/GffQECFRkO0nJ6fornheQ1UrKcXia6

2) tabindex works w/o problems - https://monosnap.com/file/KI6UhuaB66sbrrg1qFoLCsNFxWA5fL

3) added focus state for more btn and other images - https://monosnap.com/file/kmM1fUvGdJGKYYx9rxMgfOMudxGTSf

**Comment by A-Ostrovnyy** [comment]
Good job!
You must open and close your popup and inner block using the keyboard too.

**Comment by yevheniiii** [comment]
> Good job!
> You must open and close your popup and inner block using the keyboard too.

@A-Ostrovnyy 
That was a bit challenging, but I managed :) Thank you!

Here is a video of the updates - https://monosnap.com/file/aH8odclGhTKCK8h0mSFmf3qiIjecQX

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**